Frontend/CO.CDP.OrganisationApp.Tests/Pages/Registration/OrganisationDetailModelTest.cs

```diff
@@ -172,12 +176,56 @@ public void OnPost_WhenValidModel_ShouldRedirectToOrganisationDetailsPage()
     {
         var model = GivenOrganisationDetailModel();
 
+        RegistrationDetails registrationDetails = DummyRegistrationDetails();
+        sessionMock.Setup(s => s.Get<RegistrationDetails>(Session.RegistrationDetailsKey)).Returns(registrationDetails);
+
         var actionResult = model.OnPost();
 
         actionResult.Should().BeOfType<RedirectToPageResult>()
             .Which.PageName.Should().Be("OrganisationIdentification");
     }
 
+    [Fact]
+    public void OnGet_ValidSession_ReturnsRegistrationDetails()
+    {        
+        var model = GivenOrganisationDetailModel();
+
+        RegistrationDetails registrationDetails = DummyRegistrationDetails();
+
+        sessionMock.Setup(s => s.Get<RegistrationDetails>(Session.RegistrationDetailsKey)).Returns(registrationDetails);
+
+        model.OnGet();
+
+        Assert.Equal(registrationDetails.OrganisationName, model.OrganisationName);
+        Assert.Equal(registrationDetails.OrganisationType, model.OrganisationType);
+        Assert.Equal(registrationDetails.OrganisationEmailAddress, model.EmailAddress);
+        Assert.Equal(registrationDetails.OrganisationTelephoneNumber, model.TelephoneNumber);
+    }
+
+    private RegistrationDetails DummyRegistrationDetails()
+    {
+        var registrationDetails = new RegistrationDetails
+        {
+            OrganisationName = "TestOrg",
+            OrganisationType = "TestType",
+            OrganisationEmailAddress = "test@example.com",
+            OrganisationTelephoneNumber = "1234567890"
+        };
+
+        return registrationDetails;
+    }
+
+    [Fact]
+    public void OnGet_InvalidSession_ThrowsException()
+    {
+        sessionMock.Setup(s => s.Get<RegistrationDetails>(Session.RegistrationDetailsKey)).Returns<RegistrationDetails>(null);
```

Comment:
GitHub reports a warning here: `Cannot convert null literal to non-nullable reference type.`. Can we fix it?

Reply:
fixed